### ISSUE TITLE
Unable to mount volumes to ingress gateway pod using istio/gateway helm chart 

### ISSUE THREAD

**Issue by freefood89** [istio/istio#36999]
### What I Tried
Deployed istio gateway using helm chart istio/gateway (version=1.12.2) using the configs:
```yaml
values:
  podAnnotations:                                                                                       
    sidecar.istio.io/userVolumeMount: |-
      [{ "name": "google-oauth2",  "mountPath": "/etc/istio/config",  "readonly": true }]
    sidecar.istio.io/userVolume: |-
      [{ "name": "google-oauth2", "secret": { "secretName": "istio-oauth2" } }]
```

### What I thought would happen
I expected istio to mutate the pod to add `volumes`/`volumeMounts`

### What actually happened
Istio added all the usual `volumes`/`volumeMounts` but did not add the ones specified in `podAnnotations`. The `template` of the Deployment and the Pod's `annotation` do have `sidecar.istio.io/userVolumeMount` and `sidecar.istio.io/userVolume`. Why isn't the volume being injected? Is the `inject.istio.io/templates: gateway` not allowing this?

### What else I tried
- I was able to confirm that volume mounting works on ordinary sidecar injections (on httpbin) with the same annotations.
- I was able to work around this by manually deploying the gateway with `volumes`/`volumeMounts` in the Deployment. Perhaps these can be exposed on the helm chart?

### Why I need this
This is needed to mount secrets necessary for the oauth2 envoyfilter. I would like to apply this envoyfilter at the gateway level.

### Version

```prose
$ istioctl version
client version: 1.10.2
control plane version: 1.12.2
data plane version: 1.12.2 (6 proxies)

$ kubectl version --short
Client Version: v1.21.3
Server Version: v1.21.5-gke.1302

$ helm version --short
v3.5.3+g041ce5a
```

### Additional Information

(redacted gke cluster name and gcp project name)
$ istioctl bug-report

Target cluster context: ***

Running with the following config: 

istio-namespace: istio-system
full-secrets: false
timeout (mins): 30
include: {  }
exclude: { Namespaces: kube-system, kube-public, kube-node-lease, local-path-storage } AND { Namespaces: kube-system, kube-public, kube-node-lease, local-path-storage }
end-time: 2022-01-25 11:31:19.622872 -1000 HST

The following Istio control plane revisions/versions were found in the cluster:
Revision default:
&version.MeshInfo{
    {
        Component: "pilot",
        Info:      version.BuildInfo{Version:"1.12.2", GitRevision:"af0d66fd0aa363e9a7b0164f3a94ba36252fe60f", GolangVersion:"go1.17.5", BuildStatus:"Clean", GitTag:"1.12.2"},
    },
}

The following proxy revisions/versions were found in the cluster:
Revision default: Versions {1.12.2}

Fetching proxy logs for the following containers:

certmanager/cert-manager-cainjector/cert-manager-cainjector-5bcf77b697-hmz4w/cert-manager
certmanager/cert-manager-webhook/cert-manager-webhook-6f59b97fdb-ckrnj/cert-manager
certmanager/cert-manager/cert-manager-7645bbbcc9-677wk/cert-manager
cnrm-system//cnrm-controller-manager-0/manager
cnrm-system//cnrm-controller-manager-0/prom-to-sd
cnrm-system//cnrm-deletiondefender-0/deletiondefender
cnrm-system/cnrm-resource-stats-recorder/cnrm-resource-stats-recorder-7698dcb4b5-ps5h9/prom-to-sd
cnrm-system/cnrm-resource-stats-recorder/cnrm-resource-stats-recorder-7698dcb4b5-ps5h9/recorder
cnrm-system/cnrm-webhook-manager/cnrm-webhook-manager-788b44cdb7-6cqnq/webhook
cnrm-system/cnrm-webhook-manager/cnrm-webhook-manager-788b44cdb7-v4xtm/webhook
configconnector-operator-system//configconnector-operator-0/manager
default/httpbin/httpbin-54996ccdc8-r94gr/httpbin
default/httpbin/httpbin-54996ccdc8-r94gr/istio-proxy
externaldns/externaldns-external-dns/externaldns-external-dns-6bd9bd4b46-hszzm/external-dns
flux-system/helm-controller/helm-controller-779b58df6b-g7bk9/manager
flux-system/kustomize-controller/kustomize-controller-5db6bfc56d-vwxxg/manager
flux-system/notification-controller/notification-controller-7ccfbfbb98-l6xvx/manager
flux-system/source-controller/source-controller-565f8fbbff-wks2c/manager
istio-ingress/istio-ingress/istio-ingress-5d65f7878-f8gwn/istio-proxy
istio-system/istiod/istiod-5d68784db-bttp9/discovery
monitoring//alertmanager-prometheus-kube-prometheus-alertmanager-0/alertmanager
monitoring//alertmanager-prometheus-kube-prometheus-alertmanager-0/config-reloader
monitoring//alertmanager-prometheus-kube-prometheus-alertmanager-0/istio-proxy
monitoring//prometheus-prometheus-kube-prometheus-prometheus-0/config-reloader
monitoring//prometheus-prometheus-kube-prometheus-prometheus-0/istio-proxy
monitoring//prometheus-prometheus-kube-prometheus-prometheus-0/prometheus
monitoring//prometheus-prometheus-node-exporter-9hs2c/node-exporter
monitoring//prometheus-prometheus-node-exporter-jwdsl/node-exporter
monitoring//prometheus-prometheus-node-exporter-m8tr6/node-exporter
monitoring/prometheus-kube-prometheus-operator/prometheus-kube-prometheus-operator-6df74d6c5f-jlt8s/istio-proxy
monitoring/prometheus-kube-prometheus-operator/prometheus-kube-prometheus-operator-6df74d6c5f-jlt8s/kube-prometheus-stack
monitoring/prometheus-kube-state-metrics/prometheus-kube-state-metrics-79f9cf87df-nqppf/istio-proxy
monitoring/prometheus-kube-state-metrics/prometheus-kube-state-metrics-79f9cf87df-nqppf/kube-state-metrics
sealedsecrets/sealedsecrets-sealed-secrets/sealedsecrets-sealed-secrets-558d55849f-rc4m6/sealedsecrets-sealed-secrets

Fetching Istio control plane information from cluster.

The following fetches are still running: 
  kubectl describe nodes
  kubectl get secrets --all-namespaces
  kubectl describe pods -n istio-system
  kubectl get customresourcedefinitions --no-headers
  kubectl get events --all-namespaces -o wide
  kubectl get --all-namespaces all,jobs,ingresses,endpoints,customresourcedefinitions,configmaps,events -o yaml

The following fetches are still running: 
  kubectl get --all-namespaces accesscontextmanageraccesslevels.accesscontextmanager.cnrm.cloud.google.com,accesscontextmanageraccesspolicies.accesscontextmanager.cnrm.cloud.google.com,accesscontextmanagerserviceperimeters.accesscontextmanager.cnrm.cloud.google.com,alertmanagerconfigs.monitoring.coreos.com,alertmanagers.monitoring.coreos.com,alerts.notification.toolkit.fluxcd.io,artifactregistryrepositories.artifactregistry.cnrm.cloud.google.com,authorizationpolicies.security.istio.io,backendconfigs.cloud.google.com,bigquerydatasets.bigquery.cnrm.cloud.google.com,bigqueryjobs.bigquery.cnrm.cloud.google.com,bigquerytables.bigquery.cnrm.cloud.google.com,bigtableappprofiles.bigtable.cnrm.cloud.google.com,bigtablegcpolicies.bigtable.cnrm.cloud.google.com,bigtableinstances.bigtable.cnrm.cloud.google.com,bigtabletables.bigtable.cnrm.cloud.google.com,binaryauthorizationattestors.binaryauthorization.cnrm.cloud.google.com,binaryauthorizationpolicies.binaryauthorization.cnrm.cloud.google.com,buckets.source.toolkit.fluxcd.io,capacityrequests.internal.autoscaling.gke.io,certificaterequests.cert-manager.io,certificates.cert-manager.io,challenges.acme.cert-manager.io,cloudbuildtriggers.cloudbuild.cnrm.cloud.google.com,cloudidentitygroups.cloudidentity.cnrm.cloud.google.com,cloudschedulerjobs.cloudscheduler.cnrm.cloud.google.com,clusterissuers.cert-manager.io,computeaddresses.compute.cnrm.cloud.google.com,computebackendbuckets.compute.cnrm.cloud.google.com,computebackendservices.compute.cnrm.cloud.google.com,computedisks.compute.cnrm.cloud.google.com,computeexternalvpngateways.compute.cnrm.cloud.google.com,computefirewallpolicies.compute.cnrm.cloud.google.com,computefirewalls.compute.cnrm.cloud.google.com,computeforwardingrules.compute.cnrm.cloud.google.com,computehealthchecks.compute.cnrm.cloud.google.com,computehttphealthchecks.compute.cnrm.cloud.google.com,computehttpshealthchecks.compute.cnrm.cloud.google.com,computeimages.compute.cnrm.cloud.google.com,computeinstancegroupmanagers.compute.cnrm.cloud.google.com,computeinstancegroups.compute.cnrm.cloud.google.com,computeinstances.compute.cnrm.cloud.google.com,computeinstancetemplates.compute.cnrm.cloud.google.com,computeinterconnectattachments.compute.cnrm.cloud.google.com,computenetworkendpointgroups.compute.cnrm.cloud.google.com,computenetworkpeerings.compute.cnrm.cloud.google.com,computenetworks.compute.cnrm.cloud.google.com,computenodegroups.compute.cnrm.cloud.google.com,computenodetemplates.compute.cnrm.cloud.google.com,computeprojectmetadatas.compute.cnrm.cloud.google.com,computereservations.compute.cnrm.cloud.google.com,computeresourcepolicies.compute.cnrm.cloud.google.com,computerouterinterfaces.compute.cnrm.cloud.google.com,computerouternats.compute.cnrm.cloud.google.com,computerouterpeers.compute.cnrm.cloud.google.com,computerouters.compute.cnrm.cloud.google.com,computeroutes.compute.cnrm.cloud.google.com,computesecuritypolicies.compute.cnrm.cloud.google.com,computesharedvpchostprojects.compute.cnrm.cloud.google.com,computesharedvpcserviceprojects.compute.cnrm.cloud.google.com,computesnapshots.compute.cnrm.cloud.google.com,computesslcertificates.compute.cnrm.cloud.google.com,computesslpolicies.compute.cnrm.cloud.google.com,computesubnetworks.compute.cnrm.cloud.google.com,computetargetgrpcproxies.compute.cnrm.cloud.google.com,computetargethttpproxies.compute.cnrm.cloud.google.com,computetargethttpsproxies.compute.cnrm.cloud.google.com,computetargetinstances.compute.cnrm.cloud.google.com,computetargetpools.compute.cnrm.cloud.google.com,computetargetsslproxies.compute.cnrm.cloud.google.com,computetargettcpproxies.compute.cnrm.cloud.google.com,computetargetvpngateways.compute.cnrm.cloud.google.com,computeurlmaps.compute.cnrm.cloud.google.com,computevpngateways.compute.cnrm.cloud.google.com,computevpntunnels.compute.cnrm.cloud.google.com,configconnectorcontexts.core.cnrm.cloud.google.com,configconnectors.core.cnrm.cloud.google.com,containeranalysisnotes.containeranalysis.cnrm.cloud.google.com,containerclusters.container.cnrm.cloud.google.com,containernodepools.container.cnrm.cloud.google.com,dataflowflextemplatejobs.dataflow.cnrm.cloud.google.com,dataflowjobs.dataflow.cnrm.cloud.google.com,datafusioninstances.datafusion.cnrm.cloud.google.com,dataprocautoscalingpolicies.dataproc.cnrm.cloud.google.com,dataprocclusters.dataproc.cnrm.cloud.google.com,dataprocworkflowtemplates.dataproc.cnrm.cloud.google.com,destinationrules.networking.istio.io,dnsmanagedzones.dns.cnrm.cloud.google.com,dnspolicies.dns.cnrm.cloud.google.com,dnsrecordsets.dns.cnrm.cloud.google.com,envoyfilters.networking.istio.io,firestoreindexes.firestore.cnrm.cloud.google.com,folders.resourcemanager.cnrm.cloud.google.com,frontendconfigs.networking.gke.io,gameservicesrealms.gameservices.cnrm.cloud.google.com,gateways.networking.istio.io,gitrepositories.source.toolkit.fluxcd.io,gkehubfeaturememberships.gkehub.cnrm.cloud.google.com,gkehubfeatures.gkehub.cnrm.cloud.google.com,gkehubmemberships.gkehub.cnrm.cloud.google.com,helmcharts.source.toolkit.fluxcd.io,helmreleases.helm.toolkit.fluxcd.io,helmrepositories.source.toolkit.fluxcd.io,iamauditconfigs.iam.cnrm.cloud.google.com,iamcustomroles.iam.cnrm.cloud.google.com,iampartialpolicies.iam.cnrm.cloud.google.com,iampolicies.iam.cnrm.cloud.google.com,iampolicymembers.iam.cnrm.cloud.google.com,iamserviceaccountkeys.iam.cnrm.cloud.google.com,iamserviceaccounts.iam.cnrm.cloud.google.com,iapbrands.iap.cnrm.cloud.google.com,iapidentityawareproxyclients.iap.cnrm.cloud.google.com,identityplatformoauthidpconfigs.identityplatform.cnrm.cloud.google.com,identityplatformtenantoauthidpconfigs.identityplatform.cnrm.cloud.google.com,identityplatformtenants.identityplatform.cnrm.cloud.google.com,issuers.cert-manager.io,istiooperators.install.istio.io,kmscryptokeys.kms.cnrm.cloud.google.com,kmskeyrings.kms.cnrm.cloud.google.com,kustomizations.kustomize.toolkit.fluxcd.io,logginglogexclusions.logging.cnrm.cloud.google.com,logginglogsinks.logging.cnrm.cloud.google.com,managedcertificates.networking.gke.io,memcacheinstances.memcache.cnrm.cloud.google.com,monitoringalertpolicies.monitoring.cnrm.cloud.google.com,monitoringdashboards.monitoring.cnrm.cloud.google.com,monitoringgroups.monitoring.cnrm.cloud.google.com,monitoringmetricdescriptors.monitoring.cnrm.cloud.google.com,monitoringnotificationchannels.monitoring.cnrm.cloud.google.com,networksecurityauthorizationpolicies.networksecurity.cnrm.cloud.google.com,networksecurityclienttlspolicies.networksecurity.cnrm.cloud.google.com,networksecurityservertlspolicies.networksecurity.cnrm.cloud.google.com,networkservicesendpointpolicies.networkservices.cnrm.cloud.google.com,orders.acme.cert-manager.io,osconfigguestpolicies.osconfig.cnrm.cloud.google.com,peerauthentications.security.istio.io,podmonitors.monitoring.coreos.com,probes.monitoring.coreos.com,projects.resourcemanager.cnrm.cloud.google.com,prometheuses.monitoring.coreos.com,prometheusrules.monitoring.coreos.com,providers.notification.toolkit.fluxcd.io,pubsubsubscriptions.pubsub.cnrm.cloud.google.com,pubsubtopics.pubsub.cnrm.cloud.google.com,receivers.notification.toolkit.fluxcd.io,redisinstances.redis.cnrm.cloud.google.com,requestauthentications.security.istio.io,resourcemanagerliens.resourcemanager.cnrm.cloud.google.com,resourcemanagerpolicies.resourcemanager.cnrm.cloud.google.com,sealedsecrets.bitnami.com,secretmanagersecrets.secretmanager.cnrm.cloud.google.com,secretmanagersecretversions.secretmanager.cnrm.cloud.google.com,serviceattachments.networking.gke.io,serviceentries.networking.istio.io,servicemonitors.monitoring.coreos.com,servicenetworkendpointgroups.networking.gke.io,servicenetworkingconnections.servicenetworking.cnrm.cloud.google.com,services.serviceusage.cnrm.cloud.google.com,sidecars.networking.istio.io,sourcereporepositories.sourcerepo.cnrm.cloud.google.com,spannerdatabases.spanner.cnrm.cloud.google.com,spannerinstances.spanner.cnrm.cloud.google.com,sqldatabases.sql.cnrm.cloud.google.com,sqlinstances.sql.cnrm.cloud.google.com,sqlsslcerts.sql.cnrm.cloud.google.com,sqlusers.sql.cnrm.cloud.google.com,storagebucketaccesscontrols.storage.cnrm.cloud.google.com,storagebuckets.storage.cnrm.cloud.google.com,storagedefaultobjectaccesscontrols.storage.cnrm.cloud.google.com,storagenotifications.storage.cnrm.cloud.google.com,storagetransferjobs.storagetransfer.cnrm.cloud.google.com,telemetries.telemetry.istio.io,thanosrulers.monitoring.coreos.com,updateinfos.nodemanagement.gke.io,virtualservices.networking.istio.io,volumesnapshotclasses.snapshot.storage.k8s.io,volumesnapshotcontents.snapshot.storage.k8s.io,volumesnapshots.snapshot.storage.k8s.io,wasmplugins.extensions.istio.io,workloadentries.networking.istio.io,workloadgroups.networking.istio.io -o yaml

Running istio analyze on namespace configconnector-operator-system.
Running istio analyze on namespace externaldns.
Running istio analyze on namespace monitoring.
Running istio analyze on namespace sealedsecrets.
Running istio analyze on namespace certmanager.
Running istio analyze on namespace cnrm-system.
Running istio analyze on namespace istio-ingress.
Running istio analyze on namespace istio-system.
Running istio analyze on namespace default.
Running istio analyze on namespace flux-system.

Creating an archive at /Users/rmatsuka/propelai/manifests-platform/kustomize/overlays/***/***/istio/istio-ingress/bug-report.tar.gz.
Cleaning up temporary files in /var/folders/x_/3xx2j1js5cndgrj6mkpjd_rm0000gp/T/bug-report.
Done.

**Comment by jallaix** [comment]
Please reopen this issue as it forbids the use of Istio Gateway in clean GitOps processes.
EnvoyFilters (oauth2, lua) can't use values coming from Kubernetes secrets, meaning the secrets are hardcoded in the EnvoyFilters...

**Comment by niodice** [comment]
Agreed. Please re-open, this is a roadblock I recently encountered. Will it be fixed?

**Comment by phaninallamothu** [comment]
Can this be reopened please? Was stuck in the same spot.

**Comment by fams-interco** [comment]
This annotations are used by MutatingWebHook, to mount files in ingress, you will need to direct mount the volumes

**Comment by florianmutter** [comment]
Was fixed in #45894
